runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/FlinkStreamingTransformTranslators.java

```diff
@@ -347,27 +314,292 @@ public void translateNode(
       TypeInformation<WindowedValue<OutputT>> typeInfo =
           context.getTypeInfo(context.getOutput(transform));
 
-      DoFnOperator<InputT, OutputT, WindowedValue<OutputT>> doFnOperator = new DoFnOperator<>(
-          transform.getFn(),
-          new TupleTag<OutputT>("main output"),
-          Collections.<TupleTag<?>>emptyList(),
-          new DoFnOperator.DefaultOutputManagerFactory<WindowedValue<OutputT>>(),
-          windowingStrategy,
-          new HashMap<PCollectionView<?>, WindowingStrategy<?, ?>>(),
-          context.getPipelineOptions());
+      List<PCollectionView<?>> sideInputs = transform.getSideInputs();
+
+      @SuppressWarnings("unchecked")
+      PCollection<InputT> inputPCollection = (PCollection<InputT>) context.getInput(transform);
+
+      TypeInformation<WindowedValue<InputT>> inputTypeInfo =
+          context.getTypeInfo(inputPCollection);
+
+      if (sideInputs.isEmpty()) {
+        DoFnOperator<InputT, OutputT, WindowedValue<OutputT>> doFnOperator =
+            new DoFnOperator<>(
+                transform.getFn(),
+                inputTypeInfo,
+                new TupleTag<OutputT>("main output"),
+                Collections.<TupleTag<?>>emptyList(),
+                new DoFnOperator.DefaultOutputManagerFactory<WindowedValue<OutputT>>(),
+                windowingStrategy,
+                new HashMap<Integer, PCollectionView<?>>(), /* side-input mapping */
+                Collections.<PCollectionView<?>>emptyList(), /* side inputs */
+                context.getPipelineOptions());
+
+        DataStream<WindowedValue<InputT>> inputDataStream =
+            context.getInputDataStream(context.getInput(transform));
+
+        SingleOutputStreamOperator<WindowedValue<OutputT>> outDataStream = inputDataStream
+            .transform(transform.getName(), typeInfo, doFnOperator);
+
+        context.setOutputDataStream(context.getOutput(transform), outDataStream);
+      } else {
+        Tuple2<Map<Integer, PCollectionView<?>>, DataStream<RawUnionValue>> transformedSideInputs =
```

Comment:
`RawUnionValue` should probably be in `sdk.values` or actually moved to `runners-core` to be shared by this `CoGBK` implementation and the runner. It is just a trivial class, but today is mostly supposed to be internal to the `CoGBK` implementation so never should have been public (but I guess it has to for coder purposes or something).